### PR TITLE
Create an account and logging in without FB. Validations for formatting of user params. Logging out.

### PR DESCRIPTION
**Pull request by samlandfried**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,9 +1,22 @@
 class User < ApplicationRecord
+  has_secure_password validations: false
+
+  with_options if: ->(u){u.fb_id.nil?} do |user|
+    user.validate do |record|
+      record.errors.add(:password, :blank) unless record.password_digest.present?
+    end
+    user.validates_length_of :password, maximum: ActiveModel::SecurePassword::MAX_PASSWORD_LENGTH_ALLOWED
+    user.validates_confirmation_of :password, allow_blank: true
```
Comment (samlandfried) [inline app/models/user.rb:9]: Can you walk us through this stuff? I don't get it.

File: app/models/user.rb [hunk 2]
```diff
@@ -29,6 +42,20 @@ def self.from_omniauth(auth)
...
 
 private
+  def check_email_format
+    return if errors.key?(:email)
+    validates_format_of :email, with: /\A([a-z0-9_\.-]+\@[\da-z\.-]+\.[a-z\.]{2,6})\z/, message: "Address Invalid Format"
+  end
+
+  def check_phone_format
+    return if errors.key?(:phone)
+    validates_format_of :phone, with: /\d{3}[-\s]?\d{3}[-\s]?\d{4}/, message: "Number Invalid Format"
```
Comment (samlandfried) [inline app/models/user.rb:52]: This is cool. Doubt it's worth the effort, but I always appreciated those form inputs that either limit each input field to 3 or 4 characters, or automatically insert the '-' and '( )' into phone numbers.

File: app/views/shared/_user_creation_errors.html.erb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<% if @user.errors.any? %>
```
Comment (samlandfried) [inline app/views/shared/_user_creation_errors.html.erb:1]: Love these partials.

**Review by samlandfried — COMMENTED** [review]
Very cool! (PR title should be updated)

File: app/models/user.rb (hunk 1 above)
Comment (dannyradden) [reply app/models/user.rb:9]: @samlandfried has a question

**Pull request merged**